src/main/java/com/google/devtools/build/lib/analysis/platform/PlatformUtils.java

```diff
@@ -76,7 +76,7 @@ public static Platform getPlatformProto(
             ? remoteOptions.getRemoteDefaultExecProperties()
             : ImmutableSortedMap.of();
 
-    if (spawn.getExecutionPlatform() == null
+    if ((spawn.getExecutionPlatform() == null || spawn.getExecutionPlatform().execProperties().isEmpty())
```

Comment:
Without this change, `Command` protos end up populating `platform` (a deprecated field) with an empty `Platform`, increasing the size of the overall message.

In JSON it looks something like `{ "platform": {} }`.

This was caught by `//src/test/java/com/google/devtools/build/lib/remote:RemoteTests`.